### ISSUE TITLE
After updating from 0.72.1 to 0.72.2 I can't make a build (Xcode 14.2)

### ISSUE THREAD

**Issue by endyinthesun** [facebook/react-native#38294]
### Update: please refer to [this comment ](https://github.com/facebook/react-native/issues/38294#issuecomment-1631244337) for status and workarounds.

---

### Description

I run "RCT_NEW_ARCH_ENABLED=1 bundle exec pod install" and success install pods. After that, I try to make a build, but I get an error
"/Users/{user}/IdeaProjects/{project}/ios/Pods/Headers/Public/glog/glog/logging.h:512:1 Import of module 'glog.glog.log_severity' appears within namespace 'google'"

### React Native Version

0.72.2

### Output of `npx react-native info`

System:
  OS: macOS 13.4.1
  CPU: (4) x64 Intel(R) Core(TM) i5-7600K CPU @ 3.80GHz
  Memory: 23.88 MB / 24.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.10.0
    path: ~/.nvm/versions/node/v18.10.0/bin/node
  Yarn:
    version: 3.6.1
    path: ~/.yarn/bin/yarn
  npm:
    version: 9.6.7
    path: ~/.nvm/versions/node/v18.10.0/bin/npm
  Watchman:
    version: 2022.10.31.00
    path: /usr/local/bin/watchman
Managers:
  CocoaPods:
    version: 1.11.3
    path: /Users/vlad/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.2
      - iOS 16.2
      - macOS 13.1
      - tvOS 16.1
      - watchOS 9.1
  Android SDK:
    Android NDK: 23.0.7344513-beta4
IDEs:
  Android Studio: Flamingo 2022.2.1 Patch 1 Flamingo 2022.2.1 Patch 1
  Xcode:
    version: 14.2/14C18
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 11.0.17
    path: /Users/vlad/.jenv/shims/javac
  Ruby:
    version: 2.7.6
    path: /Users/vlad/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.2
    wanted: 0.72.2
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true

### Steps to reproduce

1. Run "RCT_NEW_ARCH_ENABLED=1 bundle exec pod install" and success install pods. 
2. Run "react-native run-ios" or make a build from xCode
3. Get error "/Users/{user}/IdeaProjects/{project}/ios/Pods/Headers/Public/glog/glog/logging.h:512:1 Import of module 'glog.glog.log_severity' appears within namespace 'google'"

### Snack, code example, screenshot, or link to a repository

<img width="811" alt="image" src="https://github.com/facebook/react-native/assets/43390684/23a97113-3e8f-4218-b949-6ee9a1f6630b">

**Comment by radko93** [comment]
I can confirm this issue.

**Comment by kelset** [comment]
Thanks for surfacing!
yeah it seems that it might be caused by this change in that commit by @louiszawadzki

```
  pod 'glog', :podspec => "#{prefix}/third-party-podspecs/glog.podspec", :modular_headers => true
```

I wonder why it doesn't happen with RNTester iOS 🤔 that's what we test the new arch things against.

@radko93 / @endyinthesun can either of you repro with a vanilla/react-native init'd project?

**Comment by radko93** [comment]
This actually only happened when archiving the build, not running locally.

**Comment by endyinthesun** [comment]
@kelset I tried to create a new app with "npx react-native init awesome_proj" and got the same result

**Comment by kelset** [comment]
we've been looking into this and it looks like it's something specific to Xcode 14.2 - on latest Xcode 14.3.1 everything is building fine.

The problem now is that some CIs like GitHub Actions do still default to [Xcode 14.2](https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md#xcode) in [their images](https://github.com/actions/runner-images) 😓

Current workarounds:
* upgrade your setup / CI to Xcode 14.3.1 (for example, for GitHub Actions `macos-13` images are available)
* stay on 0.72.1

We'll try to figure out a solution quickly, most likely we'll revert the commit and publish 0.72.3.

**Comment by endyinthesun** [comment]
@kelset thanks so much for your reply!

**Comment by Saravana2108** [comment]
I'm getting this issue while build the app, I'm using Xcode 14.3.1, RN version 0.72.2
<img width="1247" alt="image" src="https://github.com/facebook/react-native/assets/136422487/fa733c19-1b03-4c9d-a052-3435d5495d5b">

**Comment by fortmarek** [comment]
> I'm getting this issue while build the app, I'm using Xcode 14.3.1, RN version 0.72.2, M1 Mac

This is most likely unrelated to the bug raised in this issue that's specific to Xcode 14.2. Please, create a new issue with proper reproduction steps 🙏

**Comment by louiszawadzki** [comment]
Hi 👋 
It looks like my first commit to React Native was also my first regression 😅 

Totally fine by me to revert my change in the next version until we come up with a better solution, I believe there are way more people still using XCode 14.2 than people using the new architecture in projects with Swift modules.

**Comment by kelset** [comment]
thanks @louiszawadzki 👍

I'm testing locally that reverting the commit actually addresses the problem, after which I'll do 0.72.3 with the revert and we can look into how to address this more properly next week (once Riccardo is back).

as a sidenote, just so that I stop losing track of this link, Apple's policy is that you need to use [Xcode 14.1](https://developer.apple.com/news/upcoming-requirements/?id=04252023a) or higher atm.

**Comment by kelset** [comment]
Hey folks 👋

Update: I was able to repro locally that in the 0.72-stable branch even RNTester (on JSC+new arch) was failing to build with this problem:
<img width="1638" alt="Screenshot 2023-07-12 at 11 13 03" src="https://github.com/facebook/react-native/assets/16104054/aca5b4e6-4584-41cd-8f91-ac1e97633119">

And I was also able to verify that it was fixed by reverting the commit:
<img width="1460" alt="Screenshot 2023-07-12 at 11 17 56" src="https://github.com/facebook/react-native/assets/16104054/9d299d3a-5ecd-4120-a5d6-a23c4e9de4fe">

So I went ahead and rolled out a new patch, [0.72.3](https://github.com/facebook/react-native/releases/tag/v0.72.3), with just the revert 👍

I've also created a separate discussion to track how to better avoid having this type of regressions go unnoticed: https://github.com/react-native-community/discussions-and-proposals/discussions/687

and left a comment on the PR https://github.com/facebook/react-native/pull/38121 so that once Riccardo is back next week he has full info on the situation and we can figure out if there's a way to reapply it in a way that it can work with versions lower than 14.3.

**Comment by kelset** [comment]
@endyinthesun can you just let me know if with 0.72.3 the problem is solved for you too and we can close this?

**Comment by endyinthesun** [comment]
@kelset yes, it works, thanks =)

**Comment by numandev1** [comment]
@kelset I am using `react-native@0.72.4` and getting this error from GitHub action
```
import of module 'glog.glog.log_severity' appears within namespace 'google'

#include "glog/log_severity.h"
```
you can also check GitHub action https://github.com/numandev1/react-native-compressor/actions/runs/6139549466/job/16657563747?pr=176

**Comment by chinalwb** [comment]
If anyone comes here because of integrating [react-native-google-signin prebuild in Expo](https://react-native-google-signin.github.io/docs/setting-up/expo#rebuilding-the-app) , I did it by do NOT prebuild, and just go ahead to run `eas build -p ios --profile development `, I am using EXPO SDK 50, therefore it picks up proper XCode and do the build without a problem!

**Comment by herhu** [comment]
Hello, I know is closed but first the things is use Xcode 14.2 because is the last version compatible with the last version of MacBook Monterey (12.7.5) so which version of React Native is the compatible with this one ? or how to fix the pods in case that is the problem...

**Comment by W1ldr** [comment]
@herhu 
> Hello, I know is closed but first the things is use Xcode 14.2 because is the last version compatible with the last version of MacBook Monterey (12.7.5) so which version of React Native is the compatible with this one ?

react native 0.71.x
